library/std/src/sys/wasi/os.rs

```diff
@@ -23,13 +23,24 @@ mod libc {
     }
 }
 
-pub unsafe fn env_lock() -> impl Any {
-    cfg_if::cfg_if! {
-        if #[cfg(target_feature = "atomics")] {
-            todo!()
-        } else {
-            // No need for a lock if we're single-threaded, but this function will need
-            // to get implemented for multi-threaded scenarios
+cfg_if::cfg_if! {
+    if #[cfg(target_feature = "atomics")] {
+        // Access to the environment must be protected by a lock in multi-threaded scenarios.
+        use crate::sys_common::rwlock::StaticRwLock;
+        static ENV_LOCK: StaticRwLock = StaticRwLock::new();
```

Comment:
`StaticRwLock` has been removed (see #100581), use `sync::RwLock` instead (consequently these functions don't need to be `unsafe` anymore).